library/src/main/java/com/ricohapi/mstorage/MediaStorage.java

```diff
@@ -65,292 +63,387 @@ public void onThrowable(Throwable t) {
         });
     }
 
-    public void upload(InputStream inputStream, CompletionHandler<MediaInfo> handler) {
-        try {
-            if (accessToken == null) {
-                throw new RicohAPIException(0, "wrong usage: use the connect method to get an access token.");
+    public void upload(final InputStream inputStream, final CompletionHandler<MediaInfo> handler) {
+        authClient.getAccessToken(new CompletionHandler<AuthResult>(){
+            @Override
+            public void onCompleted(AuthResult result) {
+                try {
+                    if (result.getAccessToken() == null) {
```

Comment:
The return value of `result.getAccessToken()` is never `null` in this `onCompleted` block.  So, I don't think you need codes from L71 to L73.